posthog/types.py

```diff
@@ -102,6 +103,68 @@ class FlagsAndPayloads(TypedDict, total=True):
     featureFlagPayloads: Optional[dict[str, Any]]
 
 
+@dataclass(frozen=True)
+class FeatureFlagResult:
+    """
+    The result of calling a feature flag which includes the flag result, variant, and payload.
+    """
+
+    key: str
+    enabled: bool
+    variant: Optional[str]
+    payload: Optional[Any]
```

Comment:
seems good; is it worth including the reason here, too?  obviously we capture it but maybe (especially in the case of failure) it'd be nice to expose that to the caller.

Reply:
Yeah, do you think I should just include the reason description, or the whole reason object?

Reply:
the description string feels sufficient.